bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -524,10 +526,15 @@ void writeLac(final long ledgerId, final byte[] masterKey, final long lac, ByteB
      *          Add options
      */
     void addEntry(final long ledgerId, byte[] masterKey, final long entryId, ByteBuf toSend, WriteCallback cb,
-                  Object ctx, final int options) {
+                  Object ctx, final int options, final EnumSet<WriteFlag> writeFlags) {
         Object request = null;
         CompletionKey completionKey = null;
         if (useV2WireProtocol) {
+            if (writeFlags.contains(WriteFlag.DEFERRED_SYNC)) {
+                LOG.error("invalid writeflags {} for v2 protocol", writeFlags);
+                toSend.release();
+                cb.writeComplete(BKException.Code.IllegalOpException, ledgerId, entryId, addr, ctx);
```

Comment:
don't you need to return here?

need a test case to cover this path.

Reply:
I forgot it while doing the cut and paste from original patch.
Sure I will add a test case

Reply:
done